addons/account/models/partner.py

```diff
@@ -215,8 +215,15 @@ def _get_fiscal_position(self, partner, delivery=None):
         if not partner:
             return self.env['account.fiscal.position']
 
-        # if no delivery use invoicing
-        if not delivery:
+        company = self.env.company
+        eu_country_codes = set(self.env.ref('base.europe').country_ids.mapped('code'))
+        intra_eu = vat_exclusion = False
+        if company.vat and partner.vat:
+            intra_eu = company.vat[:2] in eu_country_codes and partner.vat[:2] in eu_country_codes
```

Comment:
```suggestion
        intra_eu = vat_exclusion = False
        if company.vat and partner.vat:
            eu_country_codes = set(self.env.ref('base.europe').country_ids.mapped('code'))
            intra_eu = company.vat[:2] in eu_country_codes and partner.vat[:2] in eu_country_codes
```
Please restrict variables definition in their scope of use as much as possible.  `ref` makes one query, and `country_ids.mapped('code')` probably makes two, unless already in cache.
Whatever can be easily avoided (while keeping the code clean) should be avoided, especially in a core method like this one ;).